src/main/kotlin/org/jmailen/gradle/kotlinter/support/EditorConfigUtils.kt

```diff
@@ -0,0 +1,55 @@
+package org.jmailen.gradle.kotlinter.support
+
+import com.pinterest.ktlint.core.KtLint
+import com.pinterest.ktlint.core.api.DefaultEditorConfigProperties
+import com.pinterest.ktlint.core.api.EditorConfigOverride
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.logging.Logger
+import org.gradle.api.provider.Property
+import java.io.File
+
+internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverride {
+    val rules = ktLintParams.disabledRules
+
+    return if (rules.isEmpty()) {
+        EditorConfigOverride.emptyEditorConfigOverride
+    } else {
+        EditorConfigOverride.from(DefaultEditorConfigProperties.disabledRulesProperty to rules.joinToString(separator = ","))
+    }
+}
+
+internal fun resetEditorconfigCacheIfNeeded(
+    wasEditorConfigChanged: Property<Boolean>,
+    logger: Logger,
+) {
+    if (wasEditorConfigChanged.get()) {
+        logger.info("Editorconfig changed, resetting KtLint caches")
+        KtLint.trimMemory() // Calling trimMemory() will also reset internal loaded `.editorconfig` cache
+    }
+}
+
+internal fun ProjectLayout.findApplicableEditorConfigFiles(): Sequence<File> {
+    val projectEditorConfig = projectDirectory.file(".editorconfig").asFile
+
+    return generateSequence(seed = projectEditorConfig) { editorconfig ->
+        if (editorconfig.isRootEditorConfig) {
+            null
+        } else {
+            editorconfig.parentFile?.parentFile?.resolve(".editorconfig")
+        }
+    }
```

Comment:
This implementation does find `.editorconfig` files located in the current directory or in any parent. Also it is nice that you stop visiting parent directories when you encounter the "root=true" property.
What I do miss however, but I might be wrong, is checking for `.editorconfig` files inside the project itself. Don't they need to be detected and listed beforehand?

Reply:
>What I do miss however, but I might be wrong, is checking for .editorconfig files inside the project itself. Don't they need to be detected and listed beforehand?

I think this is a valid comment 👀 Although I'd avoid blindly traversing through project sources looking for potential editorconfig files. I claim most project use single editorconfig defined at Gradle's `rootProject` + in advanced use cases users _override_ its settings by setting another `editorconfig` at Gradle `project` level (i.e. to have different formatting for generated files committed to the repository). 
I'll still suggest going with what I proposed here, at least for now, as I believe it addresses most cases plus most likely it'll be fixed when we switch to the new api you'll introduce for https://github.com/pinterest/ktlint/issues/1446